### PR TITLE
[docs] faq.md: update naked/bare domain section

### LINKED ISSUE #1731: [POSTFIX] do not list domain mydomain.tld in BOTH mydestination and virtual_mailbox_domains
Hi,

I do not understand I do not receive messages;
they exist on since I can send emails from their email addresses

> To :	user1@mydomain.tld [OR] user2@mydomain.tld
> From : mysender@outlook.com
> => it's Ok

> To :	mysender@outlook.com
> From :	user1@mydomain.tld, user2@mydomain.tld
> => Error
> -	550 5.1.1 <user1@mydomain.tld>: Recipient address rejected: User unknown in local recipient table 
> -	550 5.1.1 <user2@mydomain.tld>: Recipient address rejected: User unknown in local recipient table 

```
$ ./setup.sh debug show-mail-logs
Dec 23 23:22:37 mail postfix/trivial-rewrite[128551]: warning: do not list domain mydomain.tld in BOTH mydestination and virtual_mailbox_domains
Dec 23 23:22:37 mail policyd-spf[128553]: prepend Received-SPF: Pass (mailfrom) identity=mailfrom; client-ip=XXX.XXX.XXX.XXX; helo=fra01-mr2-obe.outbound.protection.outlook.com; envelope-from=mysender@outlook.com; receiver=<UNKNOWN> 
Dec 23 23:22:37 mail postfix/smtpd[128543]: NOQUEUE: reject: RCPT from mail-eopbgr90051.outbound.protection.outlook.com[XXX.XXX.XXX.XXX]: 550 5.1.1 <user1@mydomain.tld>: Recipient address rejected: User unknown in local recipient table; from=<mysender@outlook.com> to=<gers-infos@mydomain.tld> proto=ESMTP helo=<FRA01-MR2-obe.outbound.protection.outlook.com>
Dec 23 23:22:37 mail postfix/trivial-rewrite[128551]: warning: do not list domain mydomain.tld in BOTH mydestination and virtual_mailbox_domains
Dec 23 23:22:37 mail postfix/smtpd[128543]: NOQUEUE: reject: RCPT from mail-eopbgr90051.outbound.protection.outlook.com[XXX.XXX.XXX.XXX]: 550 5.1.1 <user2@mydomain.tld>: Recipient address rejected: User unknown in local recipient table; from=<mysender@outlook.com> to=<gers-news@mydomain.tld> proto=ESMTP helo=<FRA01-MR2-obe.outbound.protection.outlook.com>
Dec 23 23:22:37 mail postfix/smtpd[128543]: disconnect from mail-eopbgr90051.outbound.protection.outlook.com[XXX.XXX.XXX.XXX] ehlo=2 starttls=1 mail=1 rcpt=0/2 quit=1 commands=5/7
```

```
$ ./setup.sh email list
postmaster@mydomain.tld
user1@mydomain.tld
user2@mydomain.tld
```

```
$ cat postfix-accounts.cf 
postmaster@mydomain.tld|{SHA512-CRYPT}[...]
user1@mydomain.tld|{SHA512-CRYPT}[...]
user2@mydomain.tld|{SHA512-CRYPT}[...]
```

```
$ nano mailserver.env
[...]
# empty => all daemons start
# 1 => only launch postfix smtp
SMTP_ONLY=
[...]
```

Thank you for your help

### PR DESCRIPTION
**Pull request by sportshead**
# Description

Added a note to remove `OVERRIDE_HOSTNAME` in the env file if adding `mydestination` doesn't work. See #1731

Fixes #1731 

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by georglauterbach** [comment]
Thank you for your first contribution.

--

- [x] @polarathene I remember there were some discussions about this variable. I'm currently using it so set my hostnamen In Kubernetes. But the variable may not be perfect - is there a replacement planned or do we want to adjust the working, or domething completely different?

**Review by polarathene — DISMISSED** [review]

**Comment by polarathene** [comment]
> But the variable may not be perfect - is there a replacement planned or do we want to adjust the working, or domething completely different?

I'm unable to contribute until I restore my main PC into a working/bootable state. My replacement monitor arrived yesterday, waiting on a few more components that should arrive this week, then I should be able to attempt a restore/backup + fresh OS install. All my DMS related work is on that systems disks :sweat_smile: 

IIRC, I had notes on how to approach that and being able to run our entire test suite easily with several different configurations would help establish confidence in dropping/replacing the ENV. Without access to my notes, I can't really comment more about it :(

I only recall the use-cases for it with k8s and also host network. I will likely be prioritizing other tasks for DMS when I can contribute again, but if someone wants to look further into this topic that is welcomed :+1:

**Review by georglauterbach — APPROVED** [review]

File: docs/content/faq.md [hunk 1]
```diff
@@ -137,7 +137,7 @@ To use a bare domain (_where the host name is `example.com` and the domain is al
 - From: `mydestination = $myhostname, localhost.$mydomain, localhost`
 - To: `mydestination = localhost.$mydomain, localhost`
 
-Add the latter line to `docker-data/dms/config/postfix-main.cf`. That should work. Without that change there will be warnings in the logs like:
+Add the latter line to `docker-data/dms/config/postfix-main.cf`. If that doesn't work, make sure that `OVERRIDE_HOSTNAME` is blank in your `mailserver.env` file (see [#1731](https://github.com/docker-mailserver/docker-mailserver/issues/1731#issuecomment-753968425) Without these changes there will be warnings in the logs like:
```
Comment (casperklein) [inline docs/content/faq.md:140]:
```suggestion
Add the latter line to `docker-data/dms/config/postfix-main.cf`. If that doesn't work, make sure that `OVERRIDE_HOSTNAME` is blank in your `mailserver.env` file (see [#1731](https://github.com/docker-mailserver/docker-mailserver/issues/1731#issuecomment-753968425)). Without these changes there will be warnings in the logs like:
```

**Review by casperklein — CHANGES_REQUESTED** [review]

**Comment by sportshead** [comment]
Will force push to fix the commits

**Review by casperklein — APPROVED** [review]

**Review by georglauterbach — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**